include/rmm/aligned.hpp

```diff
@@ -16,11 +16,13 @@
 
 #pragma once
 
+#include <rmm/detail/export.hpp>
+
 #include <cassert>
 #include <cstddef>
 #include <cstdint>
 
-namespace rmm {
+namespace RMM_EXPORT rmm {
```

Comment:
Should we wrap this all up in a macro, e.g. `RMM_NAMESPACE` so that we don't have to touch every file again if we need to change the declaration of the namespace? Would simplify future PRs.

```
#define RMM_NAMESPACE RMM_EXPORT rmm
```

Reply:
Sounds reasonable

Reply:
I am going to revert the usage of `RMM_NAMESPACE` since it breaks all the doc builds since the doc engine can't see through the macro and therefore is unable to find any `rmm::` types.

Reply:
Ah, you want to do:
```diff
diff --git a/doxygen/Doxyfile b/doxygen/Doxyfile
index 434e92e0..8492b37f 100644
--- a/doxygen/Doxyfile
+++ b/doxygen/Doxyfile
@@ -2136,7 +2136,7 @@ ENABLE_PREPROCESSING   = YES
 # The default value is: NO.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-MACRO_EXPANSION        = NO
+MACRO_EXPANSION        = YES
 
 # If the EXPAND_ONLY_PREDEF and MACRO_EXPANSION tags are both set to YES then
 # the macro expansion is limited to the macros specified with the PREDEFINED and
@@ -2144,7 +2144,7 @@ MACRO_EXPANSION        = NO
 # The default value is: NO.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-EXPAND_ONLY_PREDEF     = NO
+EXPAND_ONLY_PREDEF     = YES
 
 # If the SEARCH_INCLUDES tag is set to YES, the include files in the
 # INCLUDE_PATH will be searched if a #include is found.
@@ -2176,7 +2176,7 @@ INCLUDE_FILE_PATTERNS  =
 # recursively expanded use the := operator instead of the = operator.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-PREDEFINED             =
+PREDEFINED             = RMM_NAMESPACE
 
 # If the MACRO_EXPANSION and EXPAND_ONLY_PREDEF tags are set to YES then this
 # tag can be used to specify a list of macro names that should be expanded. The
```

Reply:
Thank you @wence- . I have updated the PR with these changes